### ISSUE TITLE
Account for sRGB

### ISSUE THREAD

**Issue by retep998** [servo/servo#13378]
Currently Servo, like all other major browsers, does not account for the non-linear nature of sRGB. This means that any sort of blending or resizing operation is inherently flawed and incorrect.

The solution to this will likely involve webrender, making sure images are loaded as srgb textures so the GPU will convert to linear when sampling from them, and setting the output framebuffer to be srgb enabled, so the GPU converts back to srgb before displaying the result. Also any color constants will have to be converted from srgb to linear.

Servo has an opportunity to differentiate itself from other browsers here by actually being good at something. Don't let it go to waste.

**Comment by glennw** [comment]
Agreed, we should do this! It's not a huge amount of work (basically the steps outlined above), but it's quite fiddly to get the entire pipeline correct and ensure everything is in the correct space at the right time.

**Comment by notriddle** [comment]
Am I the only one who's worried about making some webpages look crappy because the authors already tried to compensate and we end up compensating a second time?

**Comment by retep998** [comment]
@notriddle Until someone can find an example of an actual website in the wild that compensates for the lack of sRGB compliance in modern browsers, I wouldn't be too worried about that.
